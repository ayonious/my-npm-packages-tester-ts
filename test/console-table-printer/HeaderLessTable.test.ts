import { Table } from 'console-table-printer';

describe('Example: Print a simple Table without table instance creation', () => {
  it(`simple print`, () => {
    //Create a table
    const table = new Table({
      title: 'Bundle (BUNDLENAME)',
      columns: [
        { name: 'Weapon', title: 'WEWWWW' },
        { name: 'Chroma', title: 'WEWWWW' },
        { name: 'Quality', title: '' },
        { name: 'Price' },
      ],
      charLength: { 'üëç': 2, '‚úÖ': 2 },
      style: {
        headerTop: {
          left: '',
          mid: '',
          right: '',
          other: '',
        },
        headerBottom: {
          left: '',
          mid: '',
          right: '',
          other: '',
        },
        tableBottom: {
          left: '',
          mid: '',
          right: '',
          other: '',
        },
        vertical: '',
      },
    });

    table.addRows([
      {
        Weapon: 'üëç',
        Chroma: '‚úÖ',
        Quality: 'Deluxe',
        Price: '1 11 VP',
      },
      {
        Weapon: 'thing',
        Chroma: 'some',
        Quality: 'Deluxe',
        Price: '1 123 VP',
      },
      {
        Weapon: 'asdfasdf',
        Chroma: 'asfasdf',
        Quality: 'Deluxe',
        Price: '1 1000 PP',
      },
    ]);

    expect(table.printTable()).toMatchSnapshot();
    expect(table.render()).toMatchSnapshot();
  });
});
