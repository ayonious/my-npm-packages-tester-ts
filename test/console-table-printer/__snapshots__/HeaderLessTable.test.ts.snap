// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example: Print a simple Table without table instance creation simple print 1`] = `undefined`;
